test/integration/cni/pod_egress_traffic_test.go

```diff
@@ -0,0 +1,286 @@
+			Expect(len(interfaceToPodListOnPrimaryNode.PodsOnPrimaryENI)).
+				Should(BeNumerically(">", 1))
+			Expect(len(interfaceToPodListOnPrimaryNode.PodsOnSecondaryENI)).
+				Should(BeNumerically(">", 1))
+		}
+		podsInPrimaryNode, err = f.K8sResourceManagers.PodManager().
+			GetPodsWithLabelSelector("node", "primary")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(podsInPrimaryNode.Items)).Should(BeNumerically(">", 1))
+
+		// we only need IPs form 2 PODs running in primary node for egress blocking testing cross PODs
+		podsInPrimaryNodeIPs = getPodIpsFromPodList(podsInPrimaryNode, isIPv4Cluster, 2)
+		Expect(len(podsInPrimaryNodeIPs)).Should(BeNumerically(">", 1))
+
+		By(fmt.Sprintf("creating test deployment on secondary node: %s", secondaryNode.Name))
+		secondaryNodeDeployment = manifest.
+			NewDefaultDeploymentBuilder().
+			Container(testerContainer).
+			Replicas(replicas).
+			NodeName(secondaryNode.Name).
+			PodLabel("node", "secondary").
+			Name("secondarynode-egress-tester").
+			Build()
+
+		secondaryNodeDeployment, err = f.K8sResourceManagers.
+			DeploymentManager().
+			CreateAndWaitTillDeploymentIsReady(secondaryNodeDeployment, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		if isIPv4Cluster {
+			// extra check for v4 cluster to assert pods are using both primary and secondary ENI
+			interfaceToPodListOnSecondaryNode :=
+				common.GetPodsOnPrimaryAndSecondaryInterface(secondaryNode, "node", "secondary", f)
+			// At least two Pods should be placed on the Primary and Secondary Interface
+			// on the Primary and Secondary Node in order to test all possible scenarios
+			Expect(len(interfaceToPodListOnSecondaryNode.PodsOnPrimaryENI)).
+				Should(BeNumerically(">", 1))
+			Expect(len(interfaceToPodListOnSecondaryNode.PodsOnSecondaryENI)).
+				Should(BeNumerically(">", 1))
+		}
+
+		podsInSecondaryNode, err = f.K8sResourceManagers.PodManager().
+			GetPodsWithLabelSelector("node", "secondary")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(podsInSecondaryNode.Items)).Should(BeNumerically(">", 1))
+
+		// we only need IPs form 2 PODs running in secondary node for egress blocking testing cross PODs
+		podsInSecondaryNodeIPs = getPodIpsFromPodList(podsInSecondaryNode, isIPv4Cluster, 2)
+		Expect(len(podsInSecondaryNodeIPs)).Should(BeNumerically(">", 1))
+	})
+
+	JustAfterEach(func() {
+		if primaryNodeDeployment != nil {
+			By("deleting the primary node egress-tester deployment")
+			err = f.K8sResourceManagers.DeploymentManager().
+				DeleteAndWaitTillDeploymentIsDeleted(primaryNodeDeployment)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		if secondaryNodeDeployment != nil {
+			By("deleting the secondary node egress-tester deployment")
+			err = f.K8sResourceManagers.DeploymentManager().
+				DeleteAndWaitTillDeploymentIsDeleted(secondaryNodeDeployment)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	FIt("container can access off-cluster service using egress interface", func() {
```

Comment:
nit: change "FIt" to "It" so this can be extended in the future

Reply:
yes, my error. This is for my local testing so this spec is `focused`, will remove in upcoming commits.